proto/transport.proto

```diff
@@ -0,0 +1,224 @@
+      ATTENDANT_CONTROL = 9;
+      // The transport is manually or automatically parked at a predefined floor and will not provide
+      // any further service. This usually happens in a low traffic condition for the purpose of
+      // energy saving.
+      PARKING = 10;
+      // The transport is returning to its home position, usually the lobby or ground floor.
+      HOMING = 11;
+      // Control of the transport is performed from a control panel in the control cabinet. The transport is not
+      // available for normal operation.
+      CABINET_RECALL = 12;
+      // The transport is under an occupant evacuation mode of control. This usually happens during a
+      // fire alarm when the transport is used for evacuation of occupants.
+      OCCUPANT_EVACUATION = 13;
+      // The transport is operational but is reserved for the transportation of goods only.
+      FREIGHT = 14;
+      FAILURE = 15;
+      REDUCED_SPEED = 16;
+      STORM_OPERATION = 17;
+      HIGH_WIND_OPERATION = 18;
+    }
+
+    enum Fault {
+      FAULT_UNSPECIFIED = 0;
+      CONTROLLER_FAULT = 1;
+      DRIVE_AND_MOTOR_FAULT = 2;
+      MECHANICAL_COMPONENT_FAULT = 3;
+      OVERSPEED_FAULT = 4;
+      POWER_SUPPLY_FAULT = 5;
+      SAFETY_DEVICE_FAULT = 6;
+      CONTROLLER_SUPPLY_FAULT = 7;
+      DRIVE_TEMPERATURE_EXCEEDED = 8;
+      // A comb plate safety switch is activated.
+      COMB_PLATE_FAULT = 9;
+      GENERAL_FAULT = 10;
+      DOOR_FAULT = 11;
+      LEVELLING_FAULT = 12;
+      SAFETY_CIRCUIT_BREAK_FAULT = 13;
+      FAIL_TO_START = 14;
+      ALARM_BUTTON = 15;
+      DOOR_NOT_CLOSING = 16;
+    }
+
+    // Location describes a place that the transport can stop at.
+    message Location {
+      // The id of the location.
+      string id = 1;
+      // The name of the location. e.g. 'Lobby', 'Floor 1', 'Car Park'.
+      string name = 2;
+      // Optional. Description of the location.
+      string description = 3;
+      // Optional. The floor number of the location.
+      int32 floor = 4;
+      // Optional. The attributes of the location.
+      repeated string attributes = 5;
+    }
+
+    // Journey describes the journey of the transport or payload.
+    // It will have a source and one or more destinations.
+    message Journey {
+      // The starting point of the journey.
+      Location start = 1;
+      // The destinations of the journey.
+      repeated Location destinations = 2;
+      // Optional. The reason for the journey.
+      string reason = 3;
+    }
+
+    // The current location of the transport.
+    Location current_location = 1;
+    // Optional. The next destination(s) the transport will stop at.
+    repeated Location next_destinations = 2;
+    // Optional. The direction the transport is moving.
+    Direction moving_direction = 3;
+    // Optional. The total amount of load being carried.
+    // The units this is measured in is implementation specific. See TransportSupport
+    float load = 4;
```

Comment:
When creating optional fields whose zero value could have some meaning you need to think about how to distinguish these two states (zero and absent). If possible, model the data so that zero means absent, but when that isn't possible specify the field as `optional` in the proto.